File: shared/utils/emails/nodeEmail/emailSender.ts
import * as nodemailer from "nodemailer";
import * as dotenv from "dotenv";
import * as path from "path";

dotenv.config({
    path: path.resolve(__dirname, "../../../.env")
});

async function sendEmail() {
    const transporter = nodemailer.createTransport({
        service: "gmail",
        auth: {
            user: process.env.EMAIL_USER as string,
            pass: process.env.EMAIL_PASSWORD as string
        }
    });

    await transporter.sendMail({
        from: process.env.EMAIL_USER,
        to: "chktlo003@myuct.ac.za",
        subject: "Test Email",
        text: "Hello from TypeScript"
    });

    console.log("Email sent!");
}

sendEmail().catch(console.error);

File: shared/utils/emails/pythonEmail/emailSender.py
import os
import smtplib
from email.message import EmailMessage
from dotenv import load_dotenv

load_dotenv(os.path.join(os.path.dirname(__file__), "../../../../.env"))

msg = EmailMessage()
msg["Subject"] = "Test Email"
msg["From"] = os.getenv("EMAIL_USER")
msg["To"] = "chktlo003@myuct.ac.za"
msg.set_content("This is a test pythonEmail from Python.")

with smtplib.SMTP("smtp.gmail.com", 587) as server:
    server.starttls()
    server.login(os.getenv("EMAIL_USER"), os.getenv("EMAIL_PASS"))
    server.send_message(msg)

print("Email sent!")